components/coding/studentPortal/Sidebar.tsx

```diff
@@ -25,6 +26,8 @@ export const Sidebar: React.FC<SidebarProps> = ({}: SidebarProps) => {
       setActivePage("profile");
     } else if (router.pathname.startsWith("/studentPortal/labs")) {
       setActivePage("labs");
+    } else if (router.pathname.startsWith("/studentPortal/goals")) {
+      setActivePage("goals");
```

Comment:
Setting and fetching the active page is a good candidate to move to redux.

Just creating a small redux slice to keep track of the active page and then reading it from all these different components would be a good first exercise for you to get comfortable with redux